catch/include/resource_guards.hh

```diff
@@ -219,7 +263,28 @@ class StreamGuard {
   }
 
   StreamGuard(const StreamGuard&) = delete;
-  StreamGuard(StreamGuard&&) = delete;
+
+  StreamGuard(StreamGuard&& o)
+      : stream_type_{o.stream_type_}, flags_{o.flags_}, priority_{o.priority_}, stream_{o.stream_} {
+    o.stream_type_ = Streams::nullstream;
+    o.flags_ = 0u;
+    o.priority_ = 0;
+    o.stream_ = nullptr;
```

Comment:
This entire block could simply be replacement by using the move assignment:
*this = std::move(o);

Reply:
Done.